checkstyle-configuration.xml

```diff
@@ -0,0 +1,175 @@
+<!--
+
+    Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
+    Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
+    "Sonatype" is a trademark of Sonatype, Inc.
+
+-->
+<!DOCTYPE module PUBLIC
+    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+
+  <property name="severity" value="error"/>
+
+  <property name="fileExtensions" value="java"/>
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="true"/>
+  </module>
+
+  <module name="NewlineAtEndOfFile"/>
+
+  <module name="SuppressWarningsFilter" />
+
+  <module name="TreeWalker">
+    <module name="SuppressWarningsHolder" />
+    <module name="UnnecessaryParentheses"/>
+    <module name="OuterTypeFilename"/>
+    <module name="LineLength">
+      <property name="max" value="120"/>
+      <!-- Allow longer lines in comments. -->
+      <property name="ignorePattern" value="^ \*.*"/>
+    </module>
+    <module name="UnusedImports"/>
+    <module name="AvoidStarImport"/>
```

Comment:
I believe our rules say that we can use * imports if we have enough items from the same package to import.  In practice I don't know if I've ever seen someone actually do it though.